x/auth/client/cli/tx_sign.go

```diff
@@ -303,16 +295,17 @@ func makeSignCmd() func(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return err
 		}
-		defer fp.Close()
+		if err = fp.Close(); err != nil {
```

Comment:
This is correct but it also changes behaviour. The command now fails before executing line 302.

Reply:
I think we should fail here if we are not able to properly flush the file. Otherwise it could not be written correctly and the user would think that he as it stored in a file.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/8111

Reply:
https://github.com/cosmos/cosmos-sdk/pull/8111#issuecomment-740570269 - it *DOES* return an error if the file cannot be closed.

Reply:
I know. I didn't say it _doesn't_. I say, we shouldn't print to console on error if a user main intention is to store the tx in a file (fail fast strategy).

Reply:
That's _not_ the intended original behaviour. 
The Close call was deferred and the returned error ignored exactly because you want to print a message to the console regardless. What needs to be fixed here is that the program does not return an error if Close() fails. My patch does exactly that in a Golang idiomatic way.